### ISSUE TITLE
sf-conduit and sf-conduit-delayed having excessive flight range on y

### ISSUE THREAD

**Issue by SpongeGuy** [SFort/MC-SurvivalFlight#8]
Hi SFort, it would seem that sf-conduit and sf-conduit-delayed grants conduits excessive flight range on the y-axis. I've also tested this with your other mod Better Beacon/Conduit, but I've had no luck in negating this issue. I haven't tested this with beacons.

Here's my better beacons config:
> false
^-Conduit: Add vanilla range [true] true | false
minecraft:prismarine;0.5;minecraft:prismarine_bricks;0.5;minecraft:dark_prismarine;0.5
^-Conduit: Added range per block of type [] ID;AMOUNT;...
null

It doesn't seem to matter how many blocks the conduit has around it, the range for y stays the same. For a conduit at y=66, I can fly up to y=472 until the mod kicks me out of flight.

**Comment by SpongeGuy** [comment]
I'm also having a different issue with the 'attacked' tag. I can open a new issue or elaborate here if you like.

**Comment by SFort** [comment]
this is how vanilla works if a conduit had water till y=472 it would still give you the conduit effect there if you wanna mimic conduit behavious you can also use the is wet condition or if you wanan prevent out of world stuff theres a y condition that limits max y


feel free to bundle issues as i don't have a large amout of them so it really isn't a problem

**Comment by SpongeGuy** [comment]
That's so strange. I was hoping that conduits had a spherical range on the y axis. My goal was to require players to build a chain of conduits if they want to fly up a long distance. If you have any suggestions on how I would be able to do that, let me know.

**Comment by SFort** [comment]
so what's the `attacked` issue?

**Comment by SpongeGuy** [comment]
I'm running a server with survivalflight installed and the creative_flight.script set to `attacked:40`. I'm having an issue where players who disconnect and reconnect cannot fly immediately unless they've been attacked once. When they are attacked, they can't fly for 2 seconds (which is expected), then they are able to fly for another 2 seconds until being kicked out of flight for another 2 seconds, after which the player is able to fly normally.

Basically, the player is kicked out of flight twice after being attacked once, which I believe is not what you intended when you wrote the mod.

**Comment by SpongeGuy** [comment]
Hey, I just fiddled around with conduits in vanilla, and their range does seem to be spherical on the y-axis.
I was able to be in water just above y=106 with no conduit effect using a conduit with 22 blocks and above y=119 using a conduit with 30 blocks.
![2022-02-23_10 06 31](https://user-images.githubusercontent.com/25460372/155346951-945c6a72-6d86-43e9-b0f1-60aa1e35c949.png)
![2022-02-23_10 15 23](https://user-images.githubusercontent.com/25460372/155348335-5f682301-8423-4fd4-aad9-ac9be58b9683.png)
![2022-02-23_10 07 56](https://user-images.githubusercontent.com/25460372/155346960-c325f02a-cca0-4147-a536-33bf779639af.png)

Correct me if I'm wrong, but does fabric play a part in changing this?

**Comment by SFort** [comment]
oh ye looks like i missed a check it is spherical

**Comment by SpongeGuy** [comment]
sick

**Comment by SFort** [comment]
I'm just gonna tag /sfort/fscript#6 so i remember todo that too when i do

**Comment by SpongeGuy** [comment]
You are a god SFort
